### PR TITLE
Headers modified in Socket creation.

### PR DESCRIPTION
**Pull request by bparmar-splunk**
**Update:**

- HTTP Request using format strings to insert user supplied data into the raw requests headers. So to prevent attackers to replace escape characters in request string. It is recommended to use the PrintWriter which allows us to bifurcate headers using **println** method instead of manually entering **\r\n** characters.
- Host in socket is removed as it should not be exposed to external users from request headers.

### CODE REVIEW

File: splunk/src/main/java/com/splunk/Receiver.java [hunk 1]
```diff
@@ -83,8 +83,7 @@ public Socket attach(Args args) throws IOException {
      */
     public Socket attach(String indexName, Args args) throws IOException {
         Socket socket = service.open();
-        OutputStream ostream = socket.getOutputStream();
-        Writer out = new OutputStreamWriter(ostream, "UTF-8");
+        PrintWriter writer = new PrintWriter(new OutputStreamWriter(socket.getOutputStream(), "UTF-8"));
```
Comment (fantavlik) [inline splunk/src/main/java/com/splunk/Receiver.java:86]: Why is the type of writer being changed here? It's not clear from the PR summary.

File: splunk/src/main/java/com/splunk/Receiver.java [hunk 2]
```diff
@@ -94,26 +93,19 @@ public Socket attach(String indexName, Args args) throws IOException {
...
         } else {
-            header.append("Authorization: ");
-            header.append(service.getToken());
+            headers.add(String.format("Authorization: %s", service.getToken()));
         }
-        header.append("\r\n\r\n");
-
-        out.write(header.toString());
-        out.flush();
+        headers.add("");
```
Comment (fantavlik) [inline splunk/src/main/java/com/splunk/Receiver.java:106]: Curious what this line is doing

**Review by fantavlik — CHANGES_REQUESTED** [review]

File: splunk/src/main/java/com/splunk/Receiver.java (hunk 2 above)
Comment (bparmar-splunk) [reply splunk/src/main/java/com/splunk/Receiver.java:106]: This empty string for headers is intentionally kept for terminating header string

File: splunk/src/main/java/com/splunk/Receiver.java (hunk 1 above)
Comment (bparmar-splunk) [reply splunk/src/main/java/com/splunk/Receiver.java:86]: PR summary updated.

**Review by fantavlik — APPROVED** [review]
Looks good, thanks for the explanations @bparmar-splunk 🚀

**Pull request merged**